torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2805,7 +2805,11 @@ def _constrain_range(self, a: sympy.Expr, min: int, max: int):
             if not (min <= int(a) <= max):
                 raise ValueRangeError(f"Invalid value {int(a)} for range [{min}:{max}]")
             return
-        assert isinstance(a, sympy.Symbol), "constraining non-Symbols NYI"
+        if isinstance(a, sympy.Expr) and a in self.replacements.values():
```

Comment:
I don't love the `a in self.replacements.values()` test, that's O(n) and can be potentially quite expensive when there are a lot of replacements